mayastor/src/bdev/nexus/nexus_rpc.rs

```diff
@@ -130,8 +132,13 @@ pub(crate) fn register_rpc_methods() {
             let key: Option<String> =
                 if args.key == "" { None } else { Some(args.key) };
 
+            let share_proto = ShareProtocolNexus::from_i32(args.share);
+            if share_proto.is_none() {
```

Comment:
IMHO, share_proto itself should never be None IMHO. it should always be one of the variants where::None is one of them.